### DISCUSSION TITLE
How axum provide type safe `State<T>`extractor ?

### DISCUSSION THREAD

**Discussion by nurmohammed840** [tokio-rs/axum#2005] [Q&A] [answered]
I am curious and want to know, How axum v0.6 provide [type safe `State` extractor](https://tokio.rs/blog/2022-11-25-announcing-axum-0-6-0),

I spent many hours, but could not figure it out. I would appreciate if someone could show me a minimal code example.

Edit: More specifically How does axum call `Handler` with `State<T>` ?

**Comment by davidpdrsn** [comment, marked as answer]
Most of the magic is in https://github.com/tokio-rs/axum/blob/main/axum/src/boxed.rs
